### PR TITLE
Rotation tests

### PR DESCRIPTION
**Pull request by wlsanderson**
So it turns out gravity vector is pointless and quaternions work different than I initially thought. I added the 6 test csv's that have the imu oriented on a different axis for each test. Unit tests for all 6 rotation cases were added, and all uses of gravity vector were removed from the data processor and related unit tests.

I will work on integration tests tomorrow, as well as more testing to make sure that the quaternion calculations work as intended for all axes.
Just when I start to think I understand these things, they throw curveballs...

### CODE REVIEW

**Comment by wlsanderson** [comment]
Until I can do another pass over it tomorrow, the purple nurple launch data is also invalid currently, because I used an incorrect initial quaternion in the dataset.

File: airbrakes/data_handling/data_processor.py [hunk 1]
```diff
@@ -310,7 +279,7 @@ def _calculate_vertical_velocity(self) -> npt.NDArray[np.float64]:
         vertical_accelerations = np.array(
             [
                 deadband(vertical_acceleration - GRAVITY, ACCELERATION_NOISE_THRESHOLD)
-                for vertical_acceleration in self._rotated_accelerations[self._gravity_axis_index]
+                for vertical_acceleration in self._rotated_accelerations[2]
```
Comment (JacksonElia) [inline airbrakes/data_handling/data_processor.py:282]: if we are sure you can always index it at 2, how about we make it so self._rotated_accelerations is just a numpy array (or list I guess) that just has that 3rd column. No need to do the list comprehensions and math for the other 2 columns

**Review by harshil21 — COMMENTED** [review]
So I guess none of us really understood quaternions. What I don't still understand is how the calculate_rotated_accelerations function always rotates it so that we get -9.8 on the z axis? I guess I need to read into that too.

**Comment by harshil21** [comment]
> Until I can do another pass over it tomorrow, the purple nurple launch data is also invalid currently, because I used an incorrect initial quaternion in the dataset.

When you do, can you make sure that is in #61 if that's not merged?

File: airbrakes/data_handling/data_processor.py [hunk 2]
```diff
@@ -288,14 +260,11 @@ def _calculate_rotated_accelerations(self) -> list[npt.NDArray[np.float64]]:
...
             )
-            # multiplies by gravity direction, so direction of acceleration will be the same sign regarless
-            # of IMU orientation.
-            accel_rotated_quat *= self._gravity_direction
 
             # Adds the accelerations to our list of rotated accelerations
             rotated_accelerations[0][idx] = accel_rotated_quat[1]
             rotated_accelerations[1][idx] = accel_rotated_quat[2]
-            rotated_accelerations[2][idx] = accel_rotated_quat[3]
+            rotated_accelerations[2][idx] = accel_rotated_quat[3] * -1
```
[the remark this answers is not in the document]
Comment (JacksonElia) [reply airbrakes/data_handling/data_processor.py:267]: like what I said earlier, we only care about the vertical accel now, so I would say we can probably throw out the horizontals accels.

File: airbrakes/data_handling/data_processor.py (hunk 1 above)
Comment (wlsanderson) [reply airbrakes/data_handling/data_processor.py:282]: Will do

**Comment by wlsanderson** [comment]
So I'm pretty sure I was just over-complicating the quaternions before, and it's actually makes some sense. A quaternion from the imu just represents an angle of rotation with respect to the IMU's definition of vertical (which is the -z axis). So when the imu is upside down, (+z is vertical), the associated quaternion represents basically a 180 degree rotation. So when you rotate the +z compensated acceleration [0,0,9.8] by the quaternion that represents a 180 degree rotation, you just rotate it back to the imu's definition of vertical (-z).

And this is the same for every orientation. With the gravity vector, we were assuming the quaternions rotate the vector to whichever vertical orientation we started at... but this doesn't make sense because the IMU doesn't know which orientation it is at. It only gives the rotation with respect to the -z axis.

File: tests/test_data_processor.py [hunk 3]
```diff
@@ -447,65 +480,54 @@ def test_max_altitude(self, data_processor):
...
+            ),
         ],
     )
-    def test_calculate_rotations(self, data_packets, expected_value):
+    def test_calculate_rotations(self, csv_path, expected_value, n_packets):
+        data_packets = load_data_packets(csv_path, n_packets)
         d = IMUDataProcessor()
         d.update(data_packets)
         rotations = d._rotated_accelerations
+        print(rotations)
```
[the remark this answers is not in the document]
Comment (wlsanderson) [reply tests/test_data_processor.py:522]: oh does `ruff format` not catch `print`?

**Review by JacksonElia — APPROVED** [review]
LGTM

**Pull request merged**